src/libunicode/u_str.rs

```diff
@@ -77,7 +77,7 @@ impl UnicodeStr for str {
     fn is_alphanumeric(&self) -> bool { self.chars().all(|c| c.is_alphanumeric()) }
 
     #[inline]
-    fn width(&self, is_cjk: bool) -> usize {
+    fn width(&self, is_cjk: bool) -> u32 {
```

Comment:
Maybe that calls for `u64`, for integers that do not clearly fall in the range of `u32`?